hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/DefineGroupsLayout.java

```diff
@@ -269,29 +275,38 @@ private void resetRemainingTargetsError() {
 
     private void validateRemainingTargets() {
         resetRemainingTargetsError();
-        if (targetFilter == null || (runningValidation != null && !runningValidation.isDone())) {
-            validationRequested = true;
+        if (targetFilter == null) {
             return;
         }
 
-        validationRequested = false;
-
-        final UI ui = UI.getCurrent();
-
-        runningValidation = rolloutManagement.validateTargetsInGroups(savedRolloutGroups, targetFilter,
-                System.currentTimeMillis());
-        runningValidation.addCallback(validation -> ui.access(() -> this.setGroupsValidation(validation)),
-                throwable -> ui.access(() -> this.setGroupsValidation(null)));
+        if (runningValidationsCounter.incrementAndGet() == 1) {
+            final ListenableFuture<RolloutGroupsValidation> validateTargetsInGroups = rolloutManagement
+                    .validateTargetsInGroups(savedRolloutGroups, targetFilter, System.currentTimeMillis());
+            final UI ui = UI.getCurrent();
+            validateTargetsInGroups.addCallback(validation -> ui.access(() -> setGroupsValidation(validation)),
+                    throwable -> ui.access(() -> setGroupsValidation(null)));
+            return;
+        }
 
+        runningValidationsCounter.incrementAndGet();
     }
 
-    private void setGroupsValidation(RolloutGroupsValidation validation) {
-        groupsValidation = validation;
+    /**
+     * YOU SHOULD NOT CALL THIS METHOD MANUALLY. It's only for the callback.
+     * Only 1 runningValidation should be executed. If this runningValidation is
+     * done, then this method is called. Maybe then a new runningValidation is
+     * executed.
+     * 
+     */
+    private void setGroupsValidation(final RolloutGroupsValidation validation) {
 
-        if (validationRequested) {
+        final int runningValidation = runningValidationsCounter.get();
```

Comment:
Use the benefits of AtomicInteger here: `runningValidationsCounter.getAndSet(0);` instead of resetting the counter in the next line